be/src/storage/compaction_scheduler.cpp

```diff
@@ -0,0 +1,242 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+#include "storage/compaction_scheduler.h"
+
+#include <chrono>
+#include <thread>
+
+#include "storage/compaction_manager.h"
+#include "storage/compaction_task.h"
+#include "storage/data_dir.h"
+#include "storage/storage_engine.h"
+#include "storage/tablet.h"
+
+using namespace std::chrono_literals;
+
+namespace starrocks {
+
+CompactionScheduler::CompactionScheduler() : _compaction_pool("compact_pool", config::max_compaction_task_num, 1000) {
+    CompactionManager::instance()->register_scheduler(this);
+}
+
+void CompactionScheduler::schedule() {
+    LOG(INFO) << "start compaction scheduler";
+    while (true) {
+        ++_round;
+        _wait_to_run();
+        Tablet* selected = _try_get_next_tablet();
+        if (!selected) {
+            std::unique_lock<std::mutex> lk(_mutex);
+            _cv.wait_for(lk, 10000ms);
+        } else {
+            std::shared_ptr<CompactionTask> compaction_task = selected->get_compaction(true);
+            compaction_task->set_compaction_scheduler(this);
+            compaction_task->set_task_id(CompactionManager::instance()->next_compaction_task_id());
+            PriorityThreadPool::Task task;
+            task.work_function = [compaction_task] { compaction_task->start(); };
+            LOG(INFO) << "start to run compaction."
```

Comment:
there is just some a bit difference. When a task submitted, it may be queued, not scheduled to run. so there is some time difference. Can I modify the log to submit the compaction task?

Reply:
Yeah, `submit the compaction task` is a good solution.